metadata/src/main/java/org/apache/kafka/metadata/authorizer/StandardAuthorizerData.java

```diff
@@ -329,7 +329,7 @@ private void logAuditMessage(
                 // authorized operations or a filter (e.g for regex subscriptions) to filter out authorized resources.
                 // In this case, log only at trace level.
                 if (action.logIfDenied()) {
-                    auditLog.info(buildAuditMessage(principal, requestContext, action, rule));
+                    auditLog.warn(buildAuditMessage(principal, requestContext, action, rule));
```

Comment:
The `AclAuthorizer` (as shown below) has the same implementation. This is long-standing behavior and there would have to be a strong reason to change it at this point.

>     if (authorized) {
>       // logIfAllowed is true if access is granted to the resource as a result of this authorization.
>       // In this case, log at debug level. If false, no access is actually granted, the result is used
>       // only to determine authorized operations. So log only at trace level.
>       if (action.logIfAllowed)
>         authorizerLogger.debug(logMessage)
>       else
>         authorizerLogger.trace(logMessage)
>     } else {
>       // logIfDenied is true if access to the resource was explicitly requested. Since this is an attempt
>       // to access unauthorized resources, log at info level. If false, this is either a request to determine
>       // authorized operations or a filter (e.g for regex subscriptions) to filter out authorized resources.
>       // In this case, log only at trace level.
>       if (action.logIfDenied)
>         authorizerLogger.info(logMessage)
>       else
>         authorizerLogger.trace(logMessage)
>     }

Reply:
Thanks for your reply!

Firstly, audit log is **security log**, not ~debug log~. Security log should be enabled by default during production such as fatal, error, warn, info verbosity levels. But now, we only enable **deny** log by default in KAFKA.
```java
case DENIED:
       if (action.logIfDenied()) {
              auditLog.info(......); // deny log at verbose level
        } else if (auditLog.isTraceEnabled()) {
              auditLog.trace(buildAuditMessage(principal, requestContext, action, rule)); // debug log for deny
        }
```

Secondly, the debug level is close to trace level, the `if (action.logIfAllowed)` branch is **redundant**. 
```java
case ALLOWED:
        if (action.logIfAllowed() && auditLog.isDebugEnabled()) { // redundant branch
              auditLog.debug(......); // debug log 
        } else if (auditLog.isTraceEnabled()) {
              auditLog.trace(buildAuditMessage(principal, requestContext, action, rule)); // debug log again
        }
```

In my opinion, if we don' t think "allow log" is as important as "deny log", we should **delete** the redundant branch here. 

Or we could log allow at **info** level with a switch `action.logIfAllowed = false/true` for performance consideration.
```java
    private void logAuditMessage(
        ...... ) {
        switch (rule.result()) {
            case ALLOWED:
                if (action.logIfAllowed() && auditLog.isDebugEnabled()) {
                    auditLog.debug(......); // info maybe better
                } else if (auditLog.isTraceEnabled()) {
                    auditLog.trace(buildAuditMessage(principal, requestContext, action, rule));
                }
                return;

            case DENIED:
                if (action.logIfDenied()) {
                    auditLog.info(......); // warn maybe better
                } else if (auditLog.isTraceEnabled()) {
                    auditLog.trace(buildAuditMessage(principal, requestContext, action, rule));
                }
        }
    }
```